gufe/transformation.py

```diff
@@ -0,0 +1,51 @@
+from .protocols import Protocol
+
+
+class Transformation:
+    """An edge of an alchemical network.
+
+    Connects two chemical states, with directionality.
+
+    Attributes
+    ----------
+    protocol : Protocol
+        The protocol used to perform the transformation.
+        Includes all details needed to perform calculation and encodes the
+        alchemical pathway used.
+
+    """
+
+    def __init__(
+            self,
+            protocol
+            ):
+
+        self._protocol = protocol
+
+
+    # these are ambiguous; will need a better way to consider this
+    # these could be expensive operations
+    def dg(self, estimator=None):
+        """Free energy difference of transformation based on given estimator,
+        using only data available for this edge.
+
+        """
+        ...
+
+    def ddg(self, estimator=None):
+        ...
+
+    def chemicalstate_start(self):
```

Comment:
What is meant by these?

Reply:
I think a reference to the “first” state/node were moving between

Reply:
I see, maybe being silly but wouldn't this be set in `init`?

Reply:
Yeah this is just a getter, so could be a property for example

Reply:
Yeah, these are just getters for the `ChemicalState`s at either end of the `Transformation`. Simplified naming down to `start` and `end`.